AutoMerge.jl/src/types.jl

```diff
@@ -1,7 +1,137 @@
+- `suggest_onepointzero::Bool = true`: should the AutoMerge comment include a suggestion to tag a 1.0 release for v0.x.y packages.
+- `point_to_slack::Bool = false`: should the AutoMerge comment recommend sending a message to the `#pkg-registration` Julia-Slack channel when auto-merging is not possible.
+- `registry_deps::Vector{<:AbstractString} = String[]`: list of registry dependencies, e.g your packages may depend on `General`.
+- `api_url::String = "https://api.github.com"`: the registry host API URL.
+- `check_license::Bool = false`: check package has a valid license.
+- `check_breaking_explanation::Bool = false`: Check whether the PR has release notes (collected via Registrator.jl) with a breaking change explanation.
+- `public_registries::Vector{<:AbstractString} = String[]`: If a new package registration has a UUID that matches
+  that of a package already registered in one of these registries supplied here
+  (and has either a different name or different URL) then an error will be thrown.
+  This to prevent AutoMerge from being used for "dependency confusion"
+  attacks on those registries.
+- `read_only::Bool = false`: run in read only mode.
+- `environment_variables_to_pass::Vector{<:AbstractString} = String[]`: Environment variables to pass to the subprocess that does `Pkg.add("Foo")` and `import Foo`
+"""
+Base.@kwdef struct AutoMergeConfiguration
+    merge_new_packages::Bool
+    merge_new_versions::Bool
+    new_package_waiting_period::Dates.Period
+    new_jll_package_waiting_period::Dates.Period
+    new_version_waiting_period::Dates.Period
+    new_jll_version_waiting_period::Dates.Period
+    registry::String
+    tagbot_enabled::Bool = false
+    authorized_authors::Vector{String}
+    authorized_authors_special_jll_exceptions::Vector{String}
+    additional_statuses::AbstractVector{<:AbstractString} = String[]
+    additional_check_runs::AbstractVector{<:AbstractString} = String[]
+    error_exit_if_automerge_not_applicable::Bool = false
+    master_branch::String = "master"
+    master_branch_is_default_branch::Bool = true
+    suggest_onepointzero::Bool = true
+    point_to_slack::Bool = false
+    registry_deps::Vector{<:AbstractString} = String[]
+    api_url::String = "https://api.github.com"
+    check_license::Bool = false
+    check_breaking_explanation::Bool = false
+    public_registries::Vector{<:AbstractString} = String[]
+    read_only::Bool = false
+    environment_variables_to_pass::Vector{<:AbstractString} = String[]
+end
+
+function Base.show(io::IO, ::MIME"text/plain", obj::AutoMergeConfiguration)
+    print(io, AutoMergeConfiguration, " with:")
+    for k in propertynames(obj)
+        print(io, "\n  ", k, ": `", repr(getproperty(obj, k)), "`")
+    end
+end
+
+Base.show(io::IO, ::AutoMergeConfiguration) = print(io, AutoMergeConfiguration, "(…)")
+
+
+const GENERAL_AUTOMERGE_CONFIG = AutoMergeConfiguration(
```

Comment:
values match https://github.com/JuliaRegistries/General/blob/3fa5003de7b8fa925c7d0d5fa783cafe07b3830a/.github/workflows/automerge.yml#L98-L117 except
```
merge_new_packages = true,
 merge_new_versions = true
```

are unconditional instead of based on ENV variables. We can override it with ENVs if needed in General.